src/limbo/model/gp.hpp

```diff
@@ -46,6 +46,10 @@
 #ifndef LIMBO_MODEL_GP_HPP
 #define LIMBO_MODEL_GP_HPP
 
+#ifdef _WIN32
+#include <corecrt_math_defines.h> // This brings in the M_PI define which is not in the C or C++ standard.
```

Comment:
I think I prefer the following alternative:

```c++
#define _USE_MATH_DEFINES
#include <cmath>
```

Reply:
That doesn't appear to be working for me, I get `"M_PI": identifier not found`. I think the issue is that if `<cmath>` has already been included somewhere earlier in the compilation then:
```cpp
#define _USE_MATH_DEFINES
#include <cmath>
```
will have no effect. You have to make sure that `#define _USE_MATH_DEFINES` happens before the first inclusion of `<cmath>` which can be difficult for a header-only library.

This same issue is discussed [here](https://stackoverflow.com/questions/6563810/m-pi-works-with-math-h-but-not-with-cmath-in-visual-studio) and [here](https://github.com/robotology/community/discussions/81)

How would you feel about skipping the use of the `M_PI` preprocessor constant altogether and instead use something like:
```cpp
namespace limbo {
  constexpr double PI = 3.14159265359;
  ...
```

Reply:
Most probably you need to put it here: https://github.com/resibots/limbo/blob/master/src/limbo/tools/math.hpp#L50. If this is not working, please `enabling editing by admins` for the PR and I will make the define from `waf` to be globally defined.

Reply:
> How would you feel about skipping the use of the M_PI preprocessor constant altogether and instead use something like

My issue with that is that we would need to define a bigger set of constants, otherwise we are inconsistent. And this is a Windows issue. We will find a way to by-pass it.

Reply:
putting it in `math.hpp` didn't work but putting it in `bo_base.hpp` (the only place I see where `gp.hpp` is used) did. Please see the most recent commit.

I will go ahead and enable editing on this PR.

Reply:
Thanks a lot for the effort! I will try to have a look asap!